### ISSUE TITLE
some parts of mautic are inaccessible or hard to use with screen readers

### ISSUE THREAD

**Issue by lilmike** [mautic/mautic#5864]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | [x]
| Feature request? | [ ]
| Enhancement? | [ ]

## Description:
I am blind, so use a screen reader to read my computer out loud. Some parts of mautic's web interface, however, are inaccessible or hard to use. Some examples include:

1. When selecting a radio button, you must click on the text of the option (or simulate a click, if available in the screen reader) to select the radio button.
1. In selection fields the entire field is very hard to use. For example, if choosing between disabled, rabbitmq, or beanstalkd, the only thing my screen reader sees at first is the word disabled, and then a seemingly blank text field. From experimentation I figured out that this is a (seemingly very popular) type of form field that you can type into the text field to narrow down results, and then click on the item to select. However, when simulating a click with a screen reader, the text doesn't appear to be found by it. Not to mention clicking being required is just not very accessible to begin with.
1. In dialogues that (I assume) cover the screen, such as settings dialogues, the rest of the page is still shown around it to my screen reader. So I will read the navigation menu first, then the settings section menu, and then get to the settings values. If this is indeed covering the page to sighted users, it should also hide the rest of the page from screen readers.

Note that this may or may not be a dependent library issue (I haven't looked into it enough to know if mautic does its own html output or relies on some other library). If it is a library case then switching to another, more accessible library should solve the issue (or convincing the library developers to build in accessibility), if not (or sometimes if so) then the w3c accessibility standards (aria or some such) could be used to mark up the page so screen readers can understand them.

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.12.2
| PHP version | 7.0.27

### Steps to reproduce:
1. Fire up a screen reader after navigating to a mautic page (free options include NVDA on windows, the built in voice over on mac, or orca on linux)
2. Arrow around the page to get a feel for what the screen reader is reading.
3. Press enter (or click) on options such as those provided above, and observe how the screen reader reads things, and how the screen actually looks. Also try navigating and selecting settings with just the keyboard.

**Comment by heathdutton** [comment]
These sound like library issues for the most part. Perhaps some adjustment to how the JS is fired could improve form usability. I wonder if there's a way to effectively "hide" the rest of the page when dialogues are open, by some class/attributes?

**Comment by lilmike** [comment]
Yes, I suspected it would be a library issue as well. So I don't need to go hunting around, what library does mautic use for page display and all? I would be happy to request accessibility enhancements on their side. It's very possible (unfortunately due to experience) for them to mark it as "wontfix" because they think it would take too much effort for too little reward, but in that case there are still options (though of course putting the burden on the library users to fix). The option in that case would be to use aria to mark up the page to tell screen readers how they need to interact with the controls in question. See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA for more info on aria.
Thanks for the response and positive comments :-).
-Michael.

**Comment by RCheesley** [comment]
@lilmike thanks for the feedback that you provided back in 2018 and sorry that nobody got back to you or looked into this.

Are you still using Mautic? If so I would love to hear any thoughts on how we can improve the experience for folk with visual impairments!

**Comment by npracht** [comment]
Hi, without any feedback we cannot carry this issue open anymore. I'll close it and in case a new comment justify to re-open it, I'll be glad to do it :)
